sql/parser/sql.y

```diff
@@ -1402,11 +1402,11 @@ rename_stmt:
   }
 | ALTER INDEX qualified_name RENAME TO name
   {
-    $$ = nil
+    $$ = &RenameIndex{Name: $3, NewName: Name($6), IfExists: false}
   }
```

Comment:
actually, let's put this on hold for now. the postgres namespace for indices is database-wide which is why this works, but it's really ugly.
Let's file a bug to keep track of it. We'll need to determine what extra restrictions to add on naming.